Updated Sidewalk Support (Pedestrian models, varying levels of foot traffic, etc.)

srothst1 opened the pull request:
This pull request adds additional sidewalk support to [3dstreet](https://github.com/3DStreet/3dstreet). Here is an overview of what this pull request accomplishes:
- Updates the `assets` folder with human character models found in [3dstreet-assets](https://github.com/3DStreet/3dstreet-assets). We will use these models to populate the sidewalks in our viewer. 
- Updates `3dstreet/src/aframe-streetmix-parsers.js` to parse the type `sidewalk` with the following four variants:
  - empty
  - sparse
  - normal
  - dense

  Each variant randomly adds a different number of pedestrians to the sidewalk. Some pedestrians are walking in an inbound direction and others are walking in an outbound direction. This screenshot showcases some preliminary results:
![](https://cdn.discordapp.com/attachments/906411396735049788/906411517312905236/unknown.png)
It turns out that people really bring this viewer to life :grinning: :walking_man: :rocket: 
- Updated `assets.js` to include the human character mixins.
- Updated the `dist` folder to account for the new human character assets that I added to `assets`.

***This pull request is ready for review.*** I have thought through the following topics: 
1. [FIXED] I have had some issues with GitHub permissions during the development process. I am going to take some time to ensure that this branch is fully functional. 
2. [FIXED] `human-characters-poses-2.glb` models do not have any color.
3. [FIXED] Fine-tuning the density of pedestrians for the various traffic levels. 
4. [FIXED] ~~Create some marketing content. This may be in the form of a short blog post that discusses the algorithm that I used to add the characters to the sidewalk.~~ Upon further thought, I think it would make the most sense to start with a blog post or tutorial that demonstrates and explains the synergy between streetmix and 3dstreet.

General feedback is always welcomed. Super excited about submitting my first pull request to [3dstreet](https://github.com/3DStreet/3dstreet) :fireworks:

srothst1 commented:
The models found in the file [human-characters-poses-2.blend](https://github.com/3DStreet/3dstreet-assets/blob/main/models/human-characters/human-characters-poses-2.blend) appear to be white when I open them in Blender. Surprisingly, I get the same result when opening up the models in [human-characters-poses-1.blend](https://github.com/3DStreet/3dstreet-assets/blob/main/models/human-characters/human-characters-poses-1.blend) in Blender. However, the glTF file generated from [human-characters-poses-1.blend](https://github.com/3DStreet/3dstreet-assets/blob/main/models/human-characters/human-characters-poses-1.blend) includes color and the glTF file generated from [human-characters-poses-2.blend](https://github.com/3DStreet/3dstreet-assets/blob/main/models/human-characters/human-characters-poses-2.blend) does not include color. Any suggestions for how to fix this? What am I missing here?

![image](https://user-images.githubusercontent.com/39531367/140627633-606bb356-94e7-492e-b22c-f1f8d29ecb01.png)

kfarr commented on src/aframe-streetmix-parsers.js:
```diff
@@ -151,6 +151,49 @@ function createClonedVariants (variantName, positionX, clonedObjectRadius, step
+    var variantName = "char" + String(getRandomIntInclusive(1,16));
+    var positionXYZString = getRandomArbitrary(xValueRange[0], xValueRange[1]) + ' 0 ' + zValueRange.pop();
+    var placedObjectEl = document.createElement('a-entity');
+    placedObjectEl.setAttribute('class', variantName);
```
we don't really need class name, it's redundant to mixin and we should prbly remove it in other places too

srothst1 replied:
Understood :+1: I just added a commit that removes line 187:
`placedObjectEl.setAttribute('class', variantName);`
I am happy to remove the class name from other places in `aframe-streetmix-parsers.js`, but that is probably beyond the scope of this pull request.

kfarr commented:
Thanks Sam for all of this!

I don't know why the char2.glb file didn't work the first time, but re-downloading it seems to have fixed. For Blender questions I'd first recommend watching this video: https://github.com/3DStreet/3dstreet-assets#video-tutorial

For the component logic, the distribution of pedestrians should probably fill the entire length of the street, which can vary. Different values can be provided for the `street` component `length` property. For example change line 85 in index.html to something like this to see how this looks with a short street:
`       <a-entity id="street-parent" street="length: 20" streetmix-loader set-loader-from-hash="defaultURL: https://streetmix.net/kfarr/3"></a-entity>`

srothst1 commented:
I just added a commit that addresses the issues related to the length of the street. Two important updates here:
1. Pedestrians are now added to the entire length of the sidewalk, not just one small section.
2. The number of pedestrians added for dense, normal, and sparse sidewalks is scaled based upon the length of the sidewalk. 

![image](https://user-images.githubusercontent.com/39531367/140823259-9e9abf81-547f-4abb-9855-617f6ec61b21.png)

srothst1 commented:
> I don't know why the char2.glb file didn't work the first time, but re-downloading it seems to have fixed. For Blender questions I'd first recommend watching this video: https://github.com/3DStreet/3dstreet-assets#video-tutorial

Thank you for the resource on Blender, I will take a look at the video later this afternoon. Feel free to add a commit that uploads the correct .glb file. Alternatively, you can send me the file on discord or via email and I can add it myself.

kfarr commented:
I committed the updated glb directly to this PR you should be able to pull locally and try

srothst1 commented:
Amazing! It turns out that I had to re-build the project to see the updated glTF file. Everything now looks good on my end :+1:

srothst1 commented:
@kfarr what needs to be done before this pull request is ready to be merged?

kfarr commented:
Looks good to me!
